### PR TITLE
[WIP] Feature/not fowkring

### PR DESCRIPTION
**Pull request by anz-gordonj7**
Fixes # .

Changes proposed in this pull request:
-
-

@anz-bank/sysl-developers

### CODE REVIEW

File: sysl2/sysl/cmd_template.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"path/filepath"
+
+	"github.com/anz-bank/sysl/sysl2/sysl/transforms"
+
+	"github.com/anz-bank/sysl/sysl2/sysl/parse"
+	"github.com/anz-bank/sysl/sysl2/sysl/syslutil"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type templateCmd struct {
```
Comment (springwiz) [inline sysl2/sysl/cmd_template.go]: Please add test cases.

File: sysl2/sysl/codegen.go [hunk 2]
```diff
@@ -274,34 +218,22 @@ func GenerateCode(
...
 		logger.Println(filename)
 
-		if result.GetMap() != nil {
-			codeOutput = appendCodeOutput(g, result, logger, codeOutput, filename)
-		} else if result.GetList() != nil {
-			for _, v := range result.GetList().Value {
+		if value.GetMap() != nil {
+			codeOutput = appendCodeOutput(g, value, logger, codeOutput, filename)
+		} else if value.GetList() != nil {
+			for _, v := range value.GetList().Value {
```
Comment (springwiz) [inline sysl2/sysl/codegen.go]: What happens if both the if's are false? Please add the else case.

File: sysl2/sysl/eval/exprEval.go [hunk 3]
```diff
@@ -18,7 +20,11 @@ func evalTransformStmts(txApp *sysl.Application, assign Scope, tform *sysl.Expr_
 		case *sysl.Expr_Transform_Stmt_Let:
 			logrus.Debugf("Evaluating var %s", ss.Let.Name)
 			res := Eval(txApp, assign, ss.Let.Expr)
-			assign[ss.Let.Name] = res
+			if ss.Let.Name == parse.TemplateLogString {
+				logrus.Printf("%s", res.GetS())
```
Comment (springwiz) [inline sysl2/sysl/eval/exprEval.go]: Too noisy, please use debugf.

File: sysl2/sysl/eval/unaryEval.go [hunk 4]
```diff
@@ -28,3 +32,26 @@ func unaryNeg(arg *sysl.Value) *sysl.Value {
...
+
+func unaryString(arg *sysl.Value) *sysl.Value {
+	listfn := func(items []*sysl.Value) *sysl.Value {
+		var parts []string
+		for _, item := range items {
+			parts = append(parts, unaryString(item).GetS())
+		}
+		return MakeValueString(fmt.Sprintf("[%s]", strings.Join(parts, ", ")))
+	}
+	switch x := arg.Value.(type) {
```
Comment (springwiz) [inline sysl2/sysl/eval/unaryEval.go]: Possibly panic if none of the types match.

**Comment by anzdaddy** [comment]
The PR description isn't very clear. Please elaborate.

**Pull request closed without merging**